src/unix/notbsd/linux/mod.rs

```diff
@@ -192,6 +192,31 @@ s! {
         pub if_name: *mut ::c_char,
     }
 
+    pub struct exit_status {
+        pub e_termination: ::c_short,
+        pub e_exit: ::c_short,
+    }
+
+    pub struct _timeval {
+        pub tv_sec: ::int32_t,
+        pub tv_usec: ::int32_t,
+    }
+
+    pub struct utmp {
+        pub ut_type: ::c_short,
+        pub ut_pid: ::pid_t,
+        pub ut_line: [::c_char; UT_LINESIZE],
+        pub ut_id: [::c_char; UT_IDSIZE],
+
+        pub ut_user: [::c_char; UT_NAMESIZE],
+        pub ut_host: [::c_char; UT_HOSTSIZE],
+        pub ut_exit: exit_status,
+        pub ut_session: ::int32_t,
+        pub ut_tv: _timeval,
```

Comment:
Not always.
```
pub struct timeval {
    pub tv_sec: time_t (AKA i64 on 64-bit platform),
    pub tv_usec: suseconds_t (AKA i64 on 64-bit platform),
}
```

However, according the definition of `struct utmp` on linux:
```C
struct utmp {
    ...
    ...

    /* The ut_session and ut_tv fields must be the same size when
       compiled 32- and 64-bit.  This allows data files and shared
       memory to be shared between 32- and 64-bit applications. */
    #if __WORDSIZE == 64 && defined __WORDSIZE_COMPAT32
    int32_t ut_session;           /* Session ID (getsid(2)),
                                     used for windowing */
    struct {
        int32_t tv_sec;           /* Seconds */
        int32_t tv_usec;          /* Microseconds */
    } ut_tv;                      /* Time entry was made */
    #else
    long   ut_session;           /* Session ID */
    struct timeval ut_tv;        /* Time entry was made */
    #endif

    ...
};
```
and it seems that `__WORDSIZE_COMPAT32` is usually defined on 64-bit platform, so the actual definition of `struct utmp` goes like this:
```C
struct utmp {
    ...
    ...

    int32_t ut_session;           /* Session ID (getsid(2)),
                                     used for windowing */
    struct {
        int32_t tv_sec;           /* Seconds */
        int32_t tv_usec;          /* Microseconds */
    } ut_tv;                      /* Time entry was made */

    ...
};
```

It's worth noting that getutent() reads `struct utmp`s from binary file /var/run/utmp, hence the actual size of `struct utmp` matters.